packages/lodestar/src/sync/sync.ts

```diff
@@ -199,15 +199,27 @@ export class BeaconSync implements IBeaconSync {
   private onUnknownBlockRoot = async (root: Root): Promise<void> => {
     const peerBalancer = new RoundRobinArray(this.getPeers());
     // eslint-disable-next-line no-constant-condition
-    while (true) {
+    let retry = 0;
+    const maxRetry = this.getPeers().length;
+    while (retry < maxRetry) {
       const peer = peerBalancer.next();
       if (!peer) {
         return;
       }
-      const blocks = await this.network.reqResp.beaconBlocksByRoot(peer, [root] as List<Root>);
-      if (blocks && blocks[0]) {
-        return await this.chain.receiveBlock(blocks[0]);
+      try {
+        const blocks = await this.network.reqResp.beaconBlocksByRoot(peer, [root] as List<Root>);
+        if (blocks && blocks[0]) {
+          return await this.chain.receiveBlock(blocks[0]);
+        }
+      } catch (e) {
+        this.logger.warn("Failed to get unknown block root from peer", {
```

Comment:
I think this should be verbose, otherwise we would just spam regular users with those warning which isn't any actionable item

Reply:
ok. Also I got a lot of warns like this 
```
2020-09-28 09:11:40  [NETWORK]          warn: failed to send request 444a20947a209db5 to peer 16Uiu2HAmCBZ9um5bnTWwby9n7ACmtMwfxZdaZhYQiUaMdrCUxrNx method=beacon_blocks_by_root, reason=write EPIPE
(node:18626) UnhandledPromiseRejectionWarning: Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:150:25)
    at writeGeneric (internal/stream_base_commons.js:141:3)
    at Socket._writeGeneric (net.js:768:11)
    at Socket._write (net.js:780:8)
    at doWrite (_stream_writable.js:428:12)
    at writeOrBuffer (_stream_writable.js:412:5)
    at Socket.Writable.write (_stream_writable.js:302:11)
    at /Users/tuyennguyen/Projects/workshop/lodestar/node_modules/stream-to-it/sink.js:71:20
    at async Object.sink (/Users/tuyennguyen/Projects/workshop/lodestar/node_modules/libp2p-tcp/src/socket-to-conn.js:31:9)
```
in https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/network/reqResp.ts#L339, that's why user reported this issue. Do we want to make it verbose too?

Reply:
yeah, think so, it shouldn't be unhandled rejection though :smile: